File: ts/wam/3808.ts
export const WamPnhRequestRevealAction = 3808 as const

export interface PnhRequestRevealActionEvent {
  /** field 1, wire `pnh_action` */
  pnhAction?: PNHACTIONTYPE
  /** field 2, wire `pnh_chat_party` */
  pnhChatParty?: PNHMESSAGECHATPARTY
  /** field 3, wire `pnh_chat_type` */
  pnhChatType?: PNHCHATTYPETYPE
  /** field 4, wire `pnh_entry_point` */
  pnhEntryPoint?: PNHENTRYPOINTTYPE
  /** field 5, wire `thread_id` */
  threadId?: string
}

File: rust/wam/3808.rs
pub const WAM_PNH_REQUEST_REVEAL_ACTION: u32 = 3808;

#[derive(Debug, Default)]
pub struct PnhRequestRevealActionEvent {
    /// field 1, wire `pnh_action`
    pub pnh_action: Option<PNHACTIONTYPE>,
    /// field 2, wire `pnh_chat_party`
    pub pnh_chat_party: Option<PNHMESSAGECHATPARTY>,
    /// field 3, wire `pnh_chat_type`
    pub pnh_chat_type: Option<PNHCHATTYPETYPE>,
    /// field 4, wire `pnh_entry_point`
    pub pnh_entry_point: Option<PNHENTRYPOINTTYPE>,
    /// field 5, wire `thread_id`
    pub thread_id: Option<String>,
}